### PR TITLE
fix:(multi-select) items need to be objects and add onClearAllButtonClick callback

### PR DESCRIPTION
**Pull request by carloslancha**

### CODE REVIEW

File: packages/clay-form/src/MultiSelectInput.tsx [hunk 1]
```diff
@@ -155,7 +185,16 @@ const ClayMultiSelect: React.FunctionComponent<IProps> = ({
 
 		const pastedItems = pastedText
 			.split(',')
-			.map(item => item.trim())
+			.map(itemLabel => {
+				itemLabel = itemLabel.trim();
+
+				const newItem: Item = getSourceItemByLabel(itemLabel) || {
```
Comment (matuzalemsteles) [inline packages/clay-form/src/MultiSelectInput.tsx]: Just out of curiosity, why do we need to get the item from the array of `sourceItems`?

Comment (carloslancha) [reply packages/clay-form/src/MultiSelectInput.tsx]: 'cause most of the times the value of an item is no the same as the label, so if is an item present in the autocomplete items we need to get the correct value from there. If not present we just make the value the same as the label.

Comment (matuzalemsteles) [reply packages/clay-form/src/MultiSelectInput.tsx]: Hm yeah, well thought.

File: packages/clay-form/src/MultiSelectInput.tsx [hunk 2]
```diff
@@ -115,17 +136,21 @@ const ClayMultiSelect: React.FunctionComponent<IProps> = ({
 	useLayoutEffect(() => {
 		if (sourceItems) {
```
Comment (matuzalemsteles) [inline packages/clay-form/src/MultiSelectInput.tsx]: We probably don't even need this conditional anymore.

```suggestion
		if (sourceItems.length > 0) {
```

**Comment by bryceosterhaus** [comment]
I moved multi-select to its own package for now, we can add these changes over there. https://github.com/liferay/clay/tree/master/packages/clay-multi-select

Since we still have some work to do and the deadline for release is today, Chema said to just put multi-select as an alpha still, so that we can make changes if needed

**Comment by carloslancha** [comment]
resent https://github.com/liferay/clay/pull/2544

**Pull request closed without merging**